pandas/tests/indexes/test_common.py

```diff
@@ -479,6 +479,15 @@ def test_sort_values_with_missing(index_with_missing, na_position, request):
     tm.assert_index_equal(result, expected)
 
 
+def test_sort_values_natsort_key():
+    # GH#56081
+    natsort = pytest.importorskip("natsort")
```

Comment:
Is there a way to test this without using a third party library?

Reply:
Yes. I modified the test to imitate the behaviour of `natsort` in a simple case.

`natsort` basically break strings into tuples and convert numbers. The result is a tuple of tuple, which causes this bug.